docs/handling-files.rst

```diff
@@ -0,0 +1,86 @@
+.. _handling-files:
+
+Handling Files
+================
+
+.. currentmodule:: click
+
+Click has built in features to support file and file path handling. Since files and files paths are recommended to be passed in as arguments, the examples use arguments.
```

Comment:
```suggestion
Click has built in features to support file and file path handling. The examples use arguments but the same principle applies to options as well.
```

I would not say it is a recommendation to pass files as arguments solely. It's perfectly valid to use options. `-i` and `-o` for example is commonly used for input and output.